src/service.rs

```diff
@@ -1802,6 +1838,35 @@ impl Service {
         }
     }
 
+    /// Set service launch protection.
+    /// This is a security feature that allows the service to run in a more secure environment.
+    /// there is no example because you need EV Certification and ELAM Driver to test it.
+    /// Please refer to the official documentation for more information:
+    /// <https://learn.microsoft.com/en-us/windows/win32/services/protecting-anti-malware-services->
+    pub fn set_launch_protected(&self, protection: ServiceLaunchProtected) -> crate::Result<()> {
+        let mut launch_protected =
+            unsafe { mem::zeroed::<Services::SERVICE_LAUNCH_PROTECTED_INFO>() };
+        launch_protected.dwLaunchProtected = protection.to_raw();
+        unsafe {
+            self.change_config2(
+                Services::SERVICE_CONFIG_LAUNCH_PROTECTED,
+                &mut launch_protected,
+            )
+            .map_err(Error::Winapi)
+        }
+    }
+    /// Get service launch protection.
+    /// This is a security feature that allows the service to run in a more secure environment.
+    pub fn get_launch_protected(&self) -> crate::Result<ServiceLaunchProtected> {
+        let mut data = vec![0u8; std::mem::size_of::<Services::SERVICE_LAUNCH_PROTECTED_INFO>()];
```

Comment:
No allocation is actually required here since the size is known at compile-time. You can remove `vec!` so that it forces data to be on a stack.


```suggestion
        let mut data = [0u8; std::mem::size_of::<Services::SERVICE_LAUNCH_PROTECTED_INFO>()];
```